src/main/java/seedu/address/model/stall/StallReview.java

```diff
@@ -0,0 +1,94 @@
+package seedu.address.model.stall;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.util.AppUtil;
+
+/**
+ * Represents a review of a stall in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class StallReview {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Review should only contain alphanumeric characters and spaces, and it should not be BLANK!";
+    public static final String STAR_CONSTRAINTS =
+            "Rating should only be an integer between 0-5!";
+
+    private final String review;
+    private final int star;
+
+    /**
+     * Constructs a StallReview object.
+     *
+     * @param review A valid review.
+     * @param star   An integer representing the star rating.
+     */
+    public StallReview(String review, int star) {
+        requireNonNull(review);
+        requireNonNull(star);
+        AppUtil.checkArgument(isValidReview(review), MESSAGE_CONSTRAINTS);
+        AppUtil.checkArgument(isValidStar(star), STAR_CONSTRAINTS);
+        this.review = review;
+        this.star = star;
+    }
+
+    /**
+     * Returns true if a given string is a valid review.
+     *
+     * @param test The review string to validate.
+     * @return True if the review is valid, false otherwise.
+     */
+    public static boolean isValidReview(String test) {
+        return test.matches("[\\p{Alnum}][\\p{Alnum} ]*");
```

Comment:
if reviews are allowed to contain punctuations then this needs to be changed too

Reply:
okAY MAKES SNESE